### PR TITLE
[Intl] Blacklist invalid languages

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes?
| New feature?  | yes?
| BC breaks?    | no?
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | #...   
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

see https://en.wikipedia.org/wiki/ISO_639-3#Special_codes

### CODE REVIEW

**Comment by ro0NL** [comment]
i tend to add a changelog not and go with master.. let me know.

File: src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php [hunk 1]
```diff
@@ -82,6 +82,10 @@ class LanguageDataGenerator extends AbstractDataGenerator
         'za' => 'zha',
         'zh' => 'zho',
     );
+    private static $blacklist = array(
```
[the remark this answers is not in the document]
Comment (ro0NL) [reply src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php]: both currency + region data generator already went with `private static $blacklist`, not sure we really care here :)

> Also, I know it's edgy, but "Unknown language" may be relevant for some apps.

Agree, and that's why i dont think we should apply it on lower branches. But `und` + `zxx` are not valid language codes, like "real" languages.

Now looking at https://en.wikipedia.org/wiki/ISO_639-3#Special_codes i even missed 2 :)

`mis` + `mul` should be added.

Note we do the same for other bundles, that is exclude unknown/unapplicable codes.

[the remark this answers is not in the document]
Comment (javiereguiluz) [reply src/Symfony/Component/Intl/Data/Generator/LanguageDataGenerator.php]: You are right ... we use the same generic variable name in the currency + region. It's just that I love self-explanatory variables so much!

I agree with you for not making the other changes ... and very good catch with the other wrong languages!! Thanks!

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Pull request merged**